### PR TITLE
DDOC: Some "char*" => D string conversions

### PR DESCRIPTION
**Pull request by Geod24**
Pretty straightforward if reviewed commit-by-commit.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/doc.d [hunk 1]
```diff
@@ -138,11 +137,8 @@ private class Section
             // Replace _ characters with spaces
             buf.writestring("$(DDOC_SECTION_H ");
             size_t o = buf.length;
-            for (size_t u = 0; u < namelen; u++)
-            {
-                char c = name[u];
+            foreach (c; name)
```
Comment (Geod24) [inline src/dmd/doc.d:139]: Sneaky, this does auto-decoding

File: src/dmd/doc.d [hunk 2]
```diff
@@ -108,7 +106,7 @@ private class Section
     void write(Loc loc, DocComment* dc, Scope* sc, Dsymbols* a, OutBuffer* buf)
     {
         assert(a.dim);
-        if (namelen)
+        if (!name.length)
```
Comment (kinke) [inline src/dmd/doc.d:109]: You probably didn't intend to invert the condition.

File: src/dmd/doc.d [hunk 3]
```diff
@@ -1689,16 +1685,16 @@ struct DocComment
...
     {
         if (!escapetable)
         {
             escapetable = new Escape();
             memset(escapetable, 0, Escape.sizeof);
         }
-        //printf("parseEscapes('%.*s') pescapetable = %p\n", cast(int)textlen, textstart, pescapetable);
-        const(char)* p = textstart;
-        const(char)* pend = p + textlen;
+        //printf("parseEscapes('%.*s') pescapetable = %p\n", cast(int)text.length, text, escapetable);
```
Comment (kinke) [inline src/dmd/doc.d:1695]: cast(int)text.length, text.ptr

File: src/dmd/doc.d (hunk 2 above)
Comment (Geod24) [reply src/dmd/doc.d]: Indeed, thanks

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**